### ISSUE TITLE
Flush resolver cache feature

### ISSUE THREAD

**Issue by dns2utf8** [bluejekyll/trust-dns#1608]
**Is your feature request related to a problem? Please describe.**
My application travels between many networks and some local services are sometimes know with their public or internal/local address.
That leads to confusions and clients being unable to connect because they have the internal address cached from the last network.

**Describe the solution you'd like**
I would like to have a way to flush the cache of a resolver manually, I can already detect the network change.

**Describe alternatives you've considered**
Another solution would be to set/overwrite a maximum validity of the cached records, eg. 60 seconds, so this specific application can cache 2k-10k request that are sent within that one minute before it leaves the network again.

**Additional context**
If you point me to the right files/structs I can send a PR tomorrow. So far I found this one: https://github.com/bluejekyll/trust-dns/blob/main/crates/resolver/src/caching_client.rs#L43-L45

**Comment by bluejekyll** [comment]
Does this setting suit your needs? https://github.com/bluejekyll/trust-dns/blob/main/crates/resolver/src/config.rs#L758

**Comment by dns2utf8** [comment]
This would solve part of the problem thanks!

The other is that the local dns servers may not be around after the network change. 
I am wondering if I have to reload this `trust_dns_resolver::system_conf::read_system_conf` after a network change?

**Comment by bluejekyll** [comment]
Yes, that would be interesting to figure out. I wonder if we could do it really high up the stack such that the entire Resolver structure is rebuilt. It might require a large rewrite to pull the config bits out… not sure. If you have ideas I’m open.

**Comment by dns2utf8** [comment]
On Linux, we could subscribe to NetworkManager on dbus or an inotify change of `/etc/resolve.conf` and then reload the settings on network change automatically.
Not sure how this would work on other plattforms
